appserver/web/weld-integration/src/main/java/org/glassfish/weld/BeanDeploymentArchiveImpl.java

```diff
@@ -768,7 +769,19 @@ private boolean isCDIAnnotatedClass(String className) {
     @SuppressWarnings("unchecked")
     protected BeansXml parseBeansXML(ReadableArchive archive, String beansXMLPath) throws IOException {
         URL url = getBeansXMLFileURL(archive, beansXMLPath);
-        BeansXml result =  weldBootstrap.parse(url);
+        if (WeldUtils.isEmptyBeansXmlModeALL(context)) {
+            try {
+                File beansxml = new File(url.toURI());
+                if (beansxml.length() == 0) {
+                    url = this.getClass().getResource("/beans-all.xml");
+                }
+            } catch (URISyntaxException ex) {
+                if (logger.isLoggable(Level.SEVERE)) {
+                    logger.log(Level.SEVERE, null, ex);
+                }
+            }
+        }
+        BeansXml result = weldBootstrap.parse(url);
```

Comment:
`WeldBootstrap` offers `parse(URL url, BeanDiscoveryMode emptyBeansXmlDiscoveryMode)`. Wouldn't that be simpler than mixing in synthetic `beans.xml`?